### PR TITLE
Canary feature & Little Makefile fix

### PR DESCRIPTION
**Pull request by ArnaudLcm**
## Description of the Pull Request (PR):

Firstly, this PR address a Makefile issue :

I added a variable to automatically detect the CPU architecture of host. This variable is used in :
```bash
cd $(PWD)/$(TOOLS_DIR) && curl -LO $(PROTOC_URL) && unzip protoc-24.0-linux-$(ARCHITECTURE_CPU).zip
```
This fix not found error on unzip if your architecture was not `aarch_64`.

Secondly, I added the possibility to duplicate a container as requested in Issue #130.
How does it work ?
- Copy the container's sources
- Build it

```bash
[root@m001 warewulf]# wwctl container list
  CONTAINER NAME  NODES  KERNEL VERSION                CREATION TIME        MODIFICATION TIME    SIZE     
  leap-ww         0      5.14.21-150500.55.19-default  25 Aug 23 09:21 UTC  26 Aug 22 09:22 UTC  1.4 GiB  

[root@m001 warewulf]# wwctl container cp leap-ww test
Copying sources...
Building container: test
Created image for VNFS container test: /srv/warewulf/container/test.img
Compressed image for VNFS container test: /srv/warewulf/container/test.img.gz
Container leap-ww successfully duplicated as test

[root@m001 warewulf]# wwctl container list
  CONTAINER NAME  NODES  KERNEL VERSION                CREATION TIME        MODIFICATION TIME    SIZE     
  leap-ww         0      5.14.21-150500.55.19-default  25 Aug 23 09:21 UTC  25 Aug 22 09:22 UTC  1.4 GiB  
  test            0      5.14.21-150500.55.19-default  25 Aug 23 09:22 UTC  25 Aug 22 09:24 UTC  1.4 GiB  

```

I also added an entry into the API :

```bash
[arnaud@m001 ~]$ curl -d '{"containerSource": "leap-ww", "containerDestination": "test2"}' -H "Content-Type: application/json" -X POST http://localhost:9871/v1/containercopy

{"code":2, "message":"Container leap-ww has been succesfully duplicated as test2", "details":[]}
```

### This fixes or addresses the following GitHub issues:

 - Fixes [#130](https://github.com/hpcng/warewulf/issues/130)

### CODE REVIEW

File: internal/app/wwctl/container/copy/root.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/hpcng/warewulf/internal/pkg/container"
+	"github.com/spf13/cobra"
+)
+
+var (
+	baseCmd = &cobra.Command{
+		DisableFlagsInUseLine: true,
+		Use:                   "cp CONTAINER NEW_NAME",
```
Comment (mslacken) [inline internal/app/wwctl/container/copy/root.go:11]: please rename to
```
Use: "copy CONTAINER NEW_NAME",
Aliases: []string{"cp"},
```
as all the container have long name for the commands, e.g. `delete` and not `rm` having `cp` as alias is a good alternative here.

Comment (ArnaudLcm) [reply internal/app/wwctl/container/copy/root.go]: Sure !

**Comment by ArnaudLcm** [comment]
I've pushed your last comments, should work :)

**Review by mslacken — APPROVED** [review]
Thanks.

**Comment by anderbubble** [comment]
Sorry for coming late to the conversation. There's some whitespace damage introduced in the `Makefile`, and `lscpu | grep 'Architecture' | cut -d':' -f2 | xargs` doesn't evaluate to `aarch_64` on my system, which is why I balked at implementing such a thing myself. (Not a true complaint; but that's why I hadn't already done it.)

```
$ lscpu | grep 'Architecture' | cut -d':' -f2 | xargs
aarch64
```

I'd've preferred commits to be a bit more split up. Also, they commits are missing the signoff which, in the future, will be used to assert agreement to the DCO.

https://github.com/hpcng/warewulf/pull/886

Thanks for the contribution! People really have been asking for this for a while, and I'm glad to see it in.

**Comment by ArnaudLcm** [comment]
Hello @anderbubble , thanks for feedbacks.
I was wondering what distribution are you using ?

Regarding DCO, I didn't knew your willing to switch to this system and I'll apply it to my future contributions :)

**Comment by anderbubble** [comment]
> what distribution are you using

This is a Rocky Linux 9 system running as a VM on an M1 Macbook.

Really, the problem is at `https://github.com/protocolbuffers/protobuf`, because `aarch64` is what they should be using in their releases.

**Comment by ArnaudLcm** [comment]
And you are actually right @anderbubble . 
I gave it a try on my macos too and it indeed fails. Maybe I'll try to figure out a new workaround in order to make this work on most machines because I'm not sure we are going to see in a near future a patch from protocolebuffers repository concerning this issue.

**Comment by anderbubble** [comment]
@ArnaudLcm I've included an alternative as part of https://github.com/hpcng/warewulf/pull/900, at https://github.com/hpcng/warewulf/pull/900/commits/87bcec1cb21aaedf4a6d4dcd2baefb5bf59b1e64

**Pull request merged**